subsys/modbus/modbus_serial.c

```diff
@@ -521,6 +520,8 @@ int modbus_serial_init(struct modbus_context *ctx,
 		return -ENODEV;
 	}
 
+#ifdef CONFIG_UART_USE_RUNTIME_CONFIGURE
+	struct uart_config uart_cfg;
```

Comment:
Why do you want runtime configuration to be mandatory for modbus?

Reply:
Because is is designed to use runtime configuration to e.g. update server baudrate. And there is a dependency on 'UART_USE_RUNTIME_CONFIGURE' in the Kconfig file. And based on your motivation, it must be implemented for native_posix/native_sim to do "testing modbus". Otherwise, you are not testing exactly what the test is supposed to test. Finally, I personally see no reason to have another Kconfig option.

Reply:
Just because a use case for runtime configuration exists doesn't mean it is mandatory. We have a Kconfig option to give users the power to choose if they want this or not. What gives you the right to override their choice, forcing them to include code they have explicitly said they do not want? It adds 1 KB to my STM32 build.

I run virtual system tests involving multiple zephyr native_sim applications that talk to each other using modbus. I don't want to "test modbus", I want to test my application logic.

Reply:
> Just because a use case for runtime configuration exists doesn't mean it is mandatory.

It is also because "there is a dependency on 'UART_USE_RUNTIME_CONFIGURE' in the Kconfig file" and you patch does not do what you think.

>We have a Kconfig option to give users the power to choose if they want this or not. What gives you the right to override their choice, forcing them to include code they have explicitly said they do not want?

Maybe my role as author and maintainer gives me the right to drive it and decide.
But I do not force anyone "to include code they have explicitly said they do not want". The system has this dependency either you fulfill this dependency or it does not work. Btw, Kconfig option UART_USE_RUNTIME_CONFIGURE was introduced after Modbus subsys, and other effective changes in the drivers even later.

>It adds 1 KB to my STM32 build.

Great, glad to hear it does not take that much flash. However that was not the motivation according to the commit message.

> 
> I run virtual system tests involving multiple zephyr native_sim applications that talk to each other using modbus. I don't want to "test modbus", I want to test my application logic.

This is also understandable yet. If you have implemented it for native_posix/native_sim, nothing stands in the way, right?

Reply:
> But I do not force anyone "to include code they have explicitly said they do not want". The system has this dependency either you fulfill this dependency or it does not work.

You say that, yet the modbus subsystem works just fine without it. Where is this hard dependency?

> Kconfig option UART_USE_RUNTIME_CONFIGURE was introduced after Modbus subsys, and other effective changes in the drivers even later.

That may explain why it wasn't already added. It does not explain why you won't let anyone add it now.

> > It adds 1 KB to my STM32 build.
>
> Great, glad to hear it does not take that much flash.

I don't think it is up to you or me to decide what users' code space is worth and what they should use it for. We can safely assume that anyone who chooses to disable UART_USE_RUNTIME_CONFIGURE considers the size saving to be significant.

Reply:
> > But I do not force anyone "to include code they have explicitly said they do not want". The system has this dependency either you fulfill this dependency or it does not work.
> 
> You say that, yet the modbus subsystem works just fine without it. Where is this hard dependency?

Original description as motivation for your change:
"This together with https://github.com/zephyrproject-rtos/zephyr/pull/68857 enables testing modbus communication on native_posix builds."
In the context of this project, it means being able to test tests/subsys/modbus, and requires UART driver to implement uart_configure(). Your patch had no effect because of "select UART_USE_RUNTIME_CONFIGURE". Subsys is designed to configure and reconfigure interfaces at runtime. Your changes were far from what would be reasonable or acceptable.

> > Kconfig option UART_USE_RUNTIME_CONFIGURE was introduced after Modbus subsys, and other effective changes in the drivers even later.
> 
> That may explain why it wasn't already added. It does not explain why you won't let anyone add it now.

Because these changes are not necessary and and smaller additional consumption of resources is acceptable.